src/base/io/writeCbModel.m

```diff
@@ -1,51 +1,98 @@
+% The "writeCbModel" function relies on another function
+% "io/utilities/writeSBML.m" to convert a COBRA-Matlab structure into
+% a libSBML-Matlab structure and then call libSBML to export a
+% FBCv2 file. The current version of the "writeSBML.m" does not require the
+% SBML toolbox (http://sbml.org/Software/SBMLToolbox).
+
+function outmodel = writeCbModel(model,varargin)
+%writeCbModel Write out COBRA models in various formats
 %
-% USAGE:
+% writeCbModel(model,format,fileName,varargin)
 %
 %    outmodel = writeCbModel(model, format, fileName, compSymbolList, compNameList, sbmlLevel, sbmlVersion)
-%
 % INPUTS:
 %    model:             Standard COBRA model structure
-%    format:            File format to be used ('text', 'xls', 'sbml', or 'mps')
 %
 % OPTIONAL INPUTS:
+%   format              File format to be used ('text','xls', 'mat'(default) or 'sbml')
+%                       text will only output data from required fields (with GPR rules converted to string representation)
+%                       xls is restricted to the fields defined in the xls io documentation.
 %    fileName:          File name for output file (optional, default opens
 %                       dialog box)
-%    compSymbolList:    List of compartment symbols
-%    compNameList:      List of compartment names corresponding to `compSymbolList`
-%    sbmlLevel:         SBML Level (default = 2)
-%    sbmlVersion:       SBML Version (default = 1)
+% OPTIONAL INPUTS as Parameter/Value pairs:
+% Parameter         Value
+% compSymbolList    List of compartment symbols (Cell array)
+% compNameList      List of compartment names corresponding to
+%                   compSymbolList (Cell array)
 %
 % OPTIONAL OUTPUTS:
 %    outmodel:          Only useable with sbml export. Will return the sbml structure, otherwise the input COBRA model structure is returned.
 %
 % .. Authors:
-%       - Markus Herrgard 2/5/07
 %       - Ines Thiele 01/10 - Added more options for field to write in xls format
 %       - Richard Que 3/17/10 -  Added ability to specify compartment names and symbols
 %       - Longfei Mao 26/04/2016 -  Added support for the FBCv2 format
-%
+%       - Thomas Pfau May 2017  - Changed To Parameter/Value pairs%
 % NOTE:
 %    The `writeCbModel` function relies on another function
 %    `io/utilities/writeSBML.m` to convert a COBRA-Matlab structure into
 %    a libSBML-Matlab structure and then call `libSBML` to export a
 %    FBCv2 file. The current version of the `writeSBML.m` does not require the
 %    SBML toolbox (http://sbml.org/Software/SBMLToolbox).
 
-if ~exist('compSymbolList','var') || isempty(compSymbolList)
-    compSymbolList = {'c','m','v','x','e','t','g','r','n','p','l','y'};
-    compNameList = {'Cytoplasm','Mitochondrion','Vacuole','Peroxisome','Extracellular','Pool','Golgi','Endoplasmic_reticulum','Nucleus','Periplasm','Lysosome','Glycosome'};
-end
+%For backward compatability, we are checking whether the old signature is
```

Comment:
Having a comment on the first line will break the documentation generation. Can you put this comment as an in-line comment?